core/application/impl/app_configuration_impl.cpp

```diff
@@ -1604,13 +1604,15 @@ namespace kagome::application {
     }
 
     max_parallel_downloads_ =
-        find_argument<uint32_t>(vm, "max-parallel-downloads")
-            .value_or(def_max_parallel_downloads);
+        std::max<uint32_t>(1,
+                           find_argument<uint32_t>(vm, "max-parallel-downloads")
+                               .value_or(def_max_parallel_downloads));
```

Comment:
I think we should not replace value blindly. Better to print warning for incorrect value (i.e. less then 1).